app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/BeaconScreen.kt

```diff
@@ -0,0 +1,224 @@
+import ch.epfl.cs311.wanderwave.ui.components.map.BeaconMapMarker
+import ch.epfl.cs311.wanderwave.ui.components.map.WanderwaveGoogleMap
+import ch.epfl.cs311.wanderwave.ui.components.profile.SelectImage
+import ch.epfl.cs311.wanderwave.ui.components.utils.LoadingScreen
+import ch.epfl.cs311.wanderwave.ui.theme.WanderwaveTheme
+import ch.epfl.cs311.wanderwave.viewmodel.BeaconViewModel
+import com.google.android.gms.maps.model.CameraPosition
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.CameraPositionState
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@Composable
+fun BeaconScreen(
+    beaconId: String?,
+    navigationActions: NavigationActions,
+    viewModel: BeaconViewModel = hiltViewModel()
+) {
+  LaunchedEffect(beaconId) {
+    println(beaconId)
+    if (beaconId != null) {
+      viewModel.getBeaconById(beaconId)
+    }
+  }
+
+  val uiState = viewModel.uiState.collectAsState().value
+  Column(
+      modifier = Modifier.fillMaxSize().padding(16.dp),
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        if (!uiState.isLoading) {
+          BeaconScreen(beacon = uiState.beacon!!, navigationActions = navigationActions)
+        } else {
+          LoadingScreen()
+        }
+      }
+}
+
+@Composable
+@Preview(showBackground = true)
+private fun BeaconScreenPreview() {
+  val previewBeacon =
+      Beacon(
+          id = "a",
+          location = Location(latitude = 46.519962, longitude = 6.633597, name = "EPFL"),
+          profileAndTrack =
+              listOf(
+                  ProfileTrackAssociation(
+                      Profile("e", "a", "a", 0, false, null, "1", "2"),
+                      Track("a", "Never gonna let you mn,mn,mn,mn,mn,mn,nmn,m", "Rick Astley")),
+              ),
+      )
+  val navController = rememberNavController()
+  val actions = remember(navController) { NavigationActions(navController) }
+  WanderwaveTheme { BeaconScreen(previewBeacon, actions) }
+}
+
+@Composable
+private fun BeaconScreen(beacon: Beacon, navigationActions: NavigationActions) {
+  Column(
+      modifier = Modifier.fillMaxSize().padding(8.dp).testTag("beaconScreen"),
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        BeaconInformation(beacon.location)
+        SongList(beacon, navigationActions)
+      }
+}
+
+@Composable
+fun BeaconInformation(location: Location) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(
+        text = stringResource(R.string.beaconTitle),
+        style = MaterialTheme.typography.displayLarge,
+        modifier = Modifier.testTag("beaconTitle"))
+    if (location.name.isNotBlank()) {
+      Text(
+          stringResource(R.string.beaconLocation, location.name),
+          style = MaterialTheme.typography.titleMedium,
+          modifier = Modifier.testTag("beaconLocation"))
+    }
+    // TODO: Maybe add location tracking here too?
```

Comment:
I agree, I think it's quite important for the users to still be able to see their relative position to the beacon if they want to go there